contribute/contribution-guide.md

```diff
@@ -51,9 +51,42 @@ For moderate or large contributions, you should not start coding or writing a de
 
 To avoid potential frustration during the code review cycle, we encourage you to clearly scope and design non-trivial contributions with the Beam community before you start coding.
 
-Generally, the JIRA issue is the best place to gather relevant design docs, comments, or references. It’s great to explicitly include relevant stakeholders early in the conversation. For designs that may be generally interesting, we also encourage conversations on the developer’s mailing list.
+### Initiate a Proposal
+
+Large contributions should be tracked using the `proposal` type in JIRA. This allows the Beam community to quickly identify, track, and engage on fundamental changes to the Beam project.
+
+Proposals should be created for: 
+
+1. Changes to the Beam programming model, such as new primitive transformations or semantic changes to existing transformations.
+1. Significant changes to how Beam concepts are represented in a given SDK.
+1. Changes that may require coordinated effort across multiple components, such as shared API surfaces or testing infrastructure.
```

Comment:
I agree major behavior changes to an existing runner belong here, as it affects existing users.

In my mind, I was thinking that creation of new components (runners, sdks, etc) would be handled by the feature branch process. That's a huge amount work, should be separable from what is happening in the core of the project, and won't affecting existing users or components. But it still has clear requirements for merging in to master. What do you think?